eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -1,46 +1,69 @@
 param(
   [Parameter(Mandatory = $true)]
-  $RepoOwner,
+  # Please use the RepoOwner/RepoName format: e.g. Azure/azure-sdk-for-java
+  $Repo,
   # Use this if a pull request might have been opened from one repo against another.
   # E.g Pull request opened from azure-sdk/azure-sdk prBranch --> Azure/azure-sdk baseBranch
-  $ForkRepoOwner,
-  [Parameter(Mandatory = $true)]
-  $RepoName,
+  $ForkRepoName,
   [Parameter(Mandatory = $true)]
   $BranchPrefix,
-  [Parameter(Mandatory = $true)]
-  $AuthToken
+  # Date format: Tuesday, April 12, 2022 1:36:02 PM
+  $LastCommitOlderThan
 )
 
 . (Join-Path $PSScriptRoot common.ps1)
 
-LogDebug "Operating on Repo [ $RepoName ]"
-try{
-  $branches = (Get-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref "heads/$BranchPrefix" -AuthToken $AuthToken).ref 
+LogDebug "Operating on Repo [ $Repo ]"
+if (($Repo -split "/").Count -lt 2) {
+  LogError "The $Repo does not have right format. Please use RepoOwner/RepoName. E.g. Azure/azure-sdk-for-java. Your input: $Repo"
+  exit 1
 }
-catch {
-  LogError "Get-GitHubSourceReferences failed with exception:`n$_"
+$RepoOwner = ($Repo -split "/")[0]
+$RepoName = ($Repo -split "/")[1]
+$GithubAPIBaseURI = "repos/$RepoOwner/$RepoName"
+
+function HandleGHAPICall ($api, $method="GET") {
```

Comment:
Switch to gh, as the github API has no ability to call the commit sha url directly which introduce more complexity of the script.

Reply:
We can use Invoke-RestMethod, but make the script combined with Github API helper method and rest api call. GH is flexible in this case.

Reply:
If we are switching to using gh commands directly why do we need to use the raw api calls as opposed to the higher level gh commands? This approach seems as though we are just replacing Invoke-RestMethod with gh api and I'm not sure why that is helpful.